File: src/dataset/typescript/isbn-verifier_Exercism-typescript-shybyte/claude/ClaudeSonnet4_isbn-verifier_v3.ts
function sum(array: number[]) {
    let total = 0;
    for (let i = 0; i < array.length; i++) {
        total += array[i];
    }
    return total;
}

export default class ISBN {
    private readonly isbnString: string

    public constructor(isbnString: string) {
        this.isbnString = isbnString
    }

    public isValid() {
        const cleanIsbn = this.isbnString.replace(/-/g, '');
        const length = cleanIsbn.length;
        
        if (length !== 10) {
            return false;
        }
        
        let checkSum = 0;
        
        for (let i = 0; i < length; i++) {
            const char = cleanIsbn[i];
            let digit: number;
            
            if (char === 'X' && i === 9) {
                digit = 10;
            } else if (char >= '0' && char <= '9') {
                digit = char.charCodeAt(0) - 48;
            } else {
                return false;
            }
            
            checkSum += digit * (10 - i);
        }
        
        return checkSum % 11 === 0;
    }
}

File: src/dataset/go/isbn_verifier_Exercism_go_ThomasZumsteg/claude/ClaudeSonnet4_isbn_verifier_v3.go
package isbn

func IsValidISBN(isbn string) bool {
	if len(isbn) < 10 {
		return false
	}
	
	digits, checksum := 0, 0
	for i := 0; i < len(isbn) && digits < 10; i++ {
		digit := isbn[i]
		if digit >= '0' && digit <= '9' {
			checksum += int(digit-'0') * (10 - digits)
			digits++
		} else if digit == 'X' && digits == 9 {
			checksum += 10
			digits++
		} else if digit != '-' && digit != ' ' {
			return false
		}
	}
	return digits == 10 && checksum%11 == 0
}